jetbrains-ultimate/it/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilderTest.kt

```diff
@@ -0,0 +1,202 @@
+        val fileName = "app"
+        val handlerName = "lambdaHandler"
+
+        val module = projectRule.module
+        val handler = projectRule.fixture.addLambdaHandler(subPath, fileName, handlerName)
+        projectRule.fixture.addPackageJsonFile()
+        val builtLambda = buildLambda(module, handler, Runtime.NODEJS8_10, "$subPath/$fileName.$handlerName")
+        verifyEntries(
+            builtLambda,
+            "$subPath/$fileName.js",
+            "package.json"
+        )
+        verifyPathMappings(
+            module,
+            builtLambda,
+            "%PROJECT_ROOT%" to "/",
+            "%BUILD_ROOT%" to "/"
+        )
+    }
+
+    @Test
+    fun sourceRootTakesPrecedenceOverContentRoot() {
+        val subPath = "hello_world"
+        val fileName = "app"
+        val handlerName = "lambdaHandler"
+
+        val module = projectRule.module
+        val handler = projectRule.fixture.addLambdaHandler(subPath, fileName, handlerName)
+        projectRule.fixture.addPackageJsonFile(subPath)
+
+        invokeAndWaitIfNeeded {
+            PsiTestUtil.addSourceRoot(module, handler.containingFile.virtualFile.parent)
+        }
+
+        val builtLambda = buildLambda(module, handler, Runtime.NODEJS8_10, "$fileName.$handlerName")
+        verifyEntries(
+            builtLambda,
+            "$fileName.js",
+            "package.json"
+        )
+        verifyPathMappings(
+            module,
+            builtLambda,
+            "%PROJECT_ROOT%/$subPath" to "/",
+            "%BUILD_ROOT%" to "/"
+        )
+    }
+
+    @Test
+    fun builtFromTemplate() {
+        val subPath = "hello_world"
+        val fileName = "app"
+        val handlerName = "lambdaHandler"
+        val logicalName = "SomeFunction"
+
+        projectRule.fixture.addLambdaHandler(subPath, fileName, handlerName)
+        projectRule.fixture.addPackageJsonFile(subPath)
+
+        val templateFile = projectRule.fixture.addSamTemplate(
+            logicalName = logicalName,
+            codeUri = subPath,
+            handler = "$fileName.$handlerName",
+            runtime = Runtime.NODEJS8_10
+        )
+        val templatePath = Paths.get(templateFile.virtualFile.path)
+
+        val builtLambda = buildLambdaFromTemplate(projectRule.module, templatePath, logicalName)
+
+        verifyEntries(
+            builtLambda,
+            "$fileName.js",
+            "package.json"
+        )
+        verifyPathMappings(
+            projectRule.module,
+            builtLambda,
+            "%PROJECT_ROOT%/$subPath" to "/",
+            "%BUILD_ROOT%" to "/"
+        )
+    }
+
+    @Test
+    fun dependenciesAreAdded() {
+        val subPath = "hello_world"
+        val fileName = "app"
+        val handlerName = "lambdaHandler"
+
+        val module = projectRule.module
+        val handler = projectRule.fixture.addLambdaHandler(subPath, fileName, handlerName)
+        projectRule.fixture.addPackageJsonFile(content =
```

Comment:
nit: prolly don't need the `content` parameter name here.

Reply:
Done.